iib/workers/tasks/build.py

```diff
@@ -833,20 +923,94 @@ def _adjust_operator_manifests(manifests_path):
         for pullspec in operator_csv.get_pullspecs():
             found_pullspecs.add(pullspec)
 
+    conf = get_worker_config()
+    registry_replacements = (
+        conf['iib_organization_customizations']
+        .get(organization, {})
+        .get('registry_replacements', {})
+    )
+
     # Resolve pull specs to container image digests
     replacement_pullspecs = {}
     for pullspec in found_pullspecs:
-        # Skip images that are already pinned
-        if ':' not in ImageName.parse(pullspec).tag:
-            replacement_pullspecs[pullspec] = ImageName.parse(_get_resolved_image(pullspec))
+        replacement_needed = False
+        if ':' in ImageName.parse(pullspec).tag:
+            resolved_image = ImageName.parse(pullspec)
+        else:
+            resolved_image = ImageName.parse(_get_resolved_image(pullspec))
+            replacement_needed = True
+
+        if registry_replacements.get(resolved_image.registry):
+            replacement_needed = True
+            log.debug(
+                'Replacing the registry on %s with %s',
+                pullspec,
+                registry_replacements[resolved_image.registry],
+            )
+            resolved_image.registry = registry_replacements[resolved_image.registry]
+
+        if replacement_needed:
+            replacement_pullspecs[pullspec] = resolved_image
 
     # Apply modifications to the operator bundle image metadata
     for operator_csv in operator_csvs:
         csv_file_name = os.path.basename(operator_csv.path)
-        log.info('Pinning the pull specifications on %s', csv_file_name)
+        log.info('Replacing the pull specifications on %s', csv_file_name)
         operator_csv.replace_pullspecs_everywhere(replacement_pullspecs)
 
         log.info('Setting spec.relatedImages on %s', csv_file_name)
         operator_csv.set_related_images()
 
         operator_csv.dump()
+
+    if organization:
+        _adjust_csv_annotations(manifests_path, package_name, organization)
+
+    return labels
+
+
+def _adjust_csv_annotations(manifests_path, package_name, organization):
+    """
+    Annotate ClusterServiceVersion objects based on an organization configuration.
+
+    :param str manifests_path: the full path to the directory containing the operator manifests.
+    :param str package_name: the operator package name.
+    :param str organization: the organization this bundle is for. This determines what annotations
+        to make.
+    """
+    conf = get_worker_config()
+    org_csv_annotations = (
+        conf['iib_organization_customizations'].get(organization, {}).get('csv_annotations')
+    )
+    if not org_csv_annotations:
+        log.debug('The organization %s does not have CSV annotations configured', organization)
+        return
+
+    glob_pattern = os.path.join(manifests_path, '*.clusterserviceversion.yaml')
```

Comment:
There's nothing that enforces this naming format. We must read each yaml file, `.yaml` or `.yml`, and determine which ones are CSVs. The operator-manifest library should be used here.

```
operator_manifest = OperatorManifest.from_directory(manifests_path)
for operator_csv in operator_manifest.files:     
  ...                  
```

`_adjust_operator_manifests` is already doing this.